test/e2e/volume_create_test.go

```diff
@@ -85,31 +85,44 @@ var _ = Describe("Podman volume create", func() {
 			Skip("Volume export check does not work with a remote client")
 		}
 
-		session := podmanTest.Podman([]string{"volume", "create", "myvol"})
+		volName := "my_vol_" + RandomString(10)
+		session := podmanTest.Podman([]string{"volume", "create", volName})
 		session.WaitWithDefaultTimeout()
-		volName := session.OutputToString()
 		Expect(session).Should(Exit(0))
+		Expect(session.OutputToString()).To(Equal(volName))
 
-		session = podmanTest.Podman([]string{"run", "--volume", volName + ":/data", ALPINE, "sh", "-c", "echo hello >> " + "/data/test"})
+		helloString := "hello-" + RandomString(20)
+		session = podmanTest.Podman([]string{"run", "--volume", volName + ":/data", ALPINE, "sh", "-c", "echo " + helloString + " >> /data/test"})
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(Exit(0))
 
-		check := podmanTest.Podman([]string{"volume", "export", volName})
+		// export to tar file...
+		helloTar := filepath.Join(podmanTest.TempDir, "hello.tar")
+		check := podmanTest.Podman([]string{"volume", "export", "-o", helloTar, volName})
 		check.WaitWithDefaultTimeout()
-		Expect(check.OutputToString()).To(ContainSubstring("hello"))
+		Expect(check).Should(Exit(0))
+
+		// ...then confirm that tar file has our desired content.
+		// These flags emit filename to stderr (-v), contents to stdout
+		tar := SystemExec("tar", []string{"-x", "-v", "--to-stdout", "-f", helloTar})
+		tar.WaitWithDefaultTimeout()
+		Expect(tar).To(Exit(0))
+		Expect(tar.ErrorToString()).To(Equal("test"))
+		Expect(tar.OutputToString()).To(Equal(helloString))
```

Comment:
I agree that we should not have this end up in the CI log but this change somewhat changes the purpose of the test. You know test the `--output` flag and not the default behaviour of `podman volume export`.

What we need to do is the create a new function which will not connect stderr/out to the ginkgo writer so it does not end up in the log.

Reply:
Fair enough, but "create new function" seems evil and risky and unmaintainable. Can GinkgoWriter be instrumented so it filters out NULs?

Reply:
To me, "new function" means duplication, or possibly changing interfaces. Maybe adding yet another arg to `PodmanAsUserBase()`, whose signature already has nine params. (Copy/pasting it seems like a worse option). All for one obscure corner case. It seems like a high cost/benefit.

Reply:
With all the problems I found so far adding a new argument to PodmanAsUserBase() is the least of my concerns.

Reply:
Yes that works for me, I don't mind e2e or system test as long as we know a `podman volume export` without options works.